### PR TITLE
Carleigh Crockett, finished iteration 1, almost finished iteration 2

### PR DESCRIPTION
**Pull request by Capleugh**
Couldn't figure out the end of iteration 2. Really struggled.

### CODE REVIEW

File: test/renter_test.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+class RenterTest < Minitest::Test
+  def setup
+    @jessie = Renter.new("Jessie")
+  end
+
+  def test_it_exists
+    assert_instance_of Renter, @jessie
+  end
+end
```
Comment (iandouglas) [inline test/renter_test.rb:13]: Good start here. We'd also want to test our object's attributes that get initialized to make sure our renter's name comes back successfully

File: lib/renter.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class Renter
+
+  def initialize(name)
+    @name = name
+  end
+end
```
Comment (iandouglas) [inline lib/renter.rb:6]: good start here, but we have no way to retrieve our renter's name which is needed later

File: test/apartment_test.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+require 'minitest/autorun'
+require 'minitest/pride'
+require './lib/apartment'
+require './lib/renter'
+
+class ApartmentTest < Minitest::Test
+  def setup
+    @a1 = Apartment.new({number: "A1", monthly_rent: 1200, bathrooms: 1, bedrooms: 1})
+    @jessie = Renter.new("Jessie")
+  end
```
Comment (iandouglas) [inline test/apartment_test.rb:10]: great setup

File: test/apartment_test.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+  def test_it_exists
+    assert_instance_of Apartment, @a1
+  end
+
+  def test_it_can_add_renter
+    @a1.add_renter(@jessie)
+
+    assert_equal @jessie, @a1.renter
+  end
```
Comment (iandouglas) [inline test/apartment_test.rb:20]: good test here, needed to help a little on this in class. Also, we'd want to test that our unit's renter started out `nil` in the first place

File: lib/apartment.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    @number = apartment[:number]
+    @monthly_rent = apartment[:monthly_rent]
+    @bathrooms = apartment[:bathrooms]
+    @bedrooms = apartment[:bedrooms]
+    @renter = nil
+  end
+
+  def add_renter(renter)
+    @renter = renter
+  end
```
Comment (iandouglas) [inline lib/apartment.rb:15]: good job in here, had to help a little in class

File: test/building_test.rb [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    @b2 = Apartment.new({number: "B2", monthly_rent: 999, bathrooms: 2, bedrooms: 2})
+  end
+
+  def test_it_exists
+    assert_instance_of Building, @building
+  end
+
+  def test_it_has_units
+    assert_equal [], @building.units
+  end
```
Comment (iandouglas) [inline test/building_test.rb:21]: great job here checking that the building has no units to start

File: test/building_test.rb [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+  def test_units_can_be_added
+    @building.add_unit(@a1)
+    @building.add_unit(@b2)
+
+    assert_equal [@a1, @b2], @building.units
```
Comment (iandouglas) [inline test/building_test.rb:27]: great job testing here

File: test/building_test.rb [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+  def test_average_rent
+    assert_equal ((@a1[:monthly_rent] + @a2[:monthly_rent]) / number_apartments), @building.average_rent
+  end
```
Comment (iandouglas) [inline test/building_test.rb:32]: in a case like this, we would expect that you would put the calculated value from our interaction pattern, like
`assert_equal 1099.5, @building.average_rent`
Also, `@a1` and `@b2` are your Apartment objects, and you're trying to access something called `@a1` and `@a2` here as hashes which have a key of `:monthly_rent`, which isn't how we would access our rent values.

File: lib/building.rb [hunk 9]
```diff
@@ -0,0 +1,19 @@
+class Building
+  attr_reader :units
+
+  def initialize
+    @units = []
+  end
+
+  def add_unit(unit)
+    @units << unit
+  end
```
Comment (iandouglas) [inline lib/building.rb:10]: great job up to here

File: lib/building.rb [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+
+  def average_rent
+    rent = apartments.find_all do |apartment|
+      apartment.monthly_rent
+  end
+
+  rent 
+  # this was just my best guess about where to start and I couldn't finish in time.
```
Comment (iandouglas) [inline lib/building.rb:18]: Good start! `find_all` looks for things that match a boolean expression. Technically, any unit with a rent value would evaluate as "truthy", but we'd want to collect all of those rents, add them together, and calculate an average from there.

**Review by iandouglas — COMMENTED** [review]
Good start
